.changeset/few-frogs-tease.md

```diff
@@ -0,0 +1,6 @@
+---
+'@roadiehq/backstage-plugin-github-insights': major
```

Comment:
Maybe you can wrap the exported components with the new Provider internally and export those already wrapped. That way we wouldn't need to do a breaking change.

Reply:
It wouldn't work in our case, because the state lives inside the provider. So we will need to put these providers in a place in the app where it won't get unmounted so it can keep the state when we switch pages/tabs and move around in the app

Reply:
Hmm, that's a pity. I'm not a big fan of adding new Context/Providers to the root of the app since this leaks the configuration needed by the plugin to a higher level. The whole app doesn't care about this state, only a tiny portion of our GitHub plugin does.

Could you possibly experiment with global state without Context to store this data and we can see how that looks. It would need initialization function call as well but it would be just a function call and would wrap the whole application. 

Random articles from the internet:
https://formidable.com/blog/2021/stores-no-context-api/
https://blog.openreplay.com/steps-to-develop-global-state-for-react-with-hooks-without-context

Maybe start a discussion on Discord as well for more input?

Reply:
Yea, it is :/ I don't really like it either this way, it feels wrong. I'll check out to use the global state. Other idea without provider we could use localstorage (I think it's still really hakcy). I've been reading about the HTTP 304 and I've read that the browser should use its cache for the request when we get 304, but I was not able to make it work that way initially, I'll check again also this one.